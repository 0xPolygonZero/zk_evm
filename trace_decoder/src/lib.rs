#![feature(linked_list_cursors)]
#![feature(trait_alias)]
#![feature(iter_array_chunks)]
// TODO: address these lints
#![allow(unused)]
#![allow(private_interfaces)]

mod compact;
pub mod decoding;
mod deserializers;
pub mod processed_block_trace;
pub mod trace_protocol;
pub mod types;
pub mod utils;
