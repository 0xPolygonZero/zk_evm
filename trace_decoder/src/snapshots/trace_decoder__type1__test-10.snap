---
source: trace_decoder/src/lib.rs
expression: executed
---
NonEmpty {
    inner: [
        Leaf(
            Leaf {
                key: NonEmpty {
                    inner: [
                        1,
                        8,
                        11,
                        2,
                        8,
                        9,
                        9,
                        3,
                        6,
                        10,
                        0,
                        8,
                        7,
                        4,
                        12,
                        12,
                        12,
                        14,
                        14,
                        6,
                        5,
                        7,
                        1,
                        2,
                        12,
                        8,
                        8,
                        12,
                        13,
                        10,
                        10,
                        0,
                        10,
                        3,
                        0,
                        5,
                        11,
                        0,
                        0,
                        4,
                        13,
                        3,
                        15,
                        13,
                        10,
                        2,
                        9,
                        4,
                        2,
                        11,
                        2,
                        11,
                        2,
                        13,
                        12,
                        5,
                        4,
                        15,
                        1,
                        4,
                        15,
                        6,
                        1,
                        1,
                    ],
                },
                value: Right(
                    Account {
                        nonce: 0,
                        balance: 1000000000,
                        storage: Some(
                            Leaf(
                                Leaf {
                                    key: NonEmpty {
                                        inner: [
                                            8,
                                            0,
                                            1,
                                            5,
                                            6,
                                            5,
                                            7,
                                            14,
                                            2,
                                            9,
                                            8,
                                            13,
                                            3,
                                            5,
                                            2,
                                            9,
                                            0,
                                            14,
                                            6,
                                            9,
                                            6,
                                            2,
                                            8,
                                            11,
                                            14,
                                            0,
                                            3,
                                            13,
                                            9,
                                            1,
                                            15,
                                            7,
                                            4,
                                            13,
                                            6,
                                            1,
                                            3,
                                            12,
                                            10,
                                            15,
                                            3,
                                            10,
                                            15,
                                            13,
                                            11,
                                            14,
                                            0,
                                            9,
                                            1,
                                            3,
                                            8,
                                            12,
                                            15,
                                            10,
                                            4,
                                            1,
                                            5,
                                            14,
                                            15,
                                            14,
                                            2,
                                            15,
                                            5,
                                            0,
                                        ],
                                    },
                                    value: Left(
                                        Value {
                                            raw_value: NonEmpty {
                                                inner: [
                                                    222,
                                                    173,
                                                    190,
                                                    239,
                                                ],
                                            },
                                        },
                                    ),
                                },
                            ),
                        ),
                        code: Some(
                            Left(
                                Hash {
                                    raw_hash: [
                                        166,
                                        136,
                                        91,
                                        55,
                                        49,
                                        112,
                                        45,
                                        166,
                                        46,
                                        142,
                                        74,
                                        143,
                                        88,
                                        74,
                                        196,
                                        106,
                                        127,
                                        104,
                                        34,
                                        244,
                                        226,
                                        186,
                                        80,
                                        251,
                                        169,
                                        2,
                                        246,
                                        123,
                                        21,
                                        136,
                                        210,
                                        59,
                                    ],
                                },
                            ),
                        ),
                    },
                ),
            },
        ),
    ],
}
